versioned_docs/version-4.6/developers/applications/caching.md

```diff
@@ -1,292 +1,85 @@
-
-```javascript
-class ThirdPartyAPI extends Resource {
- async get() {
-  const context = this.getContext();
-  let headers = new Headers();
-  if (context.replacingVersion) // this is the existing cached record
-   headers.set('If-Modified-Since', new Date(context.replacingVersion).toUTCString());
-  let response = await fetch(`https://some-api.com/${this.getId()}`, { headers });
-  let cacheInfo = response.headers.get('Cache-Control');
-  let maxAge = cacheInfo?.match(/max-age=(\d)/)?.[1];
-  if (maxAge) // we can set a specific expiration time by setting context.expiresAt
-   context.expiresAt = Date.now() + maxAge * 1000; // convert from seconds to milliseconds and add to current time
-  // we can just revalidate and return the record if the origin has confirmed that it has the same version:
-  if (response.status === 304) return context.replacingRecord;
-  ...
-```
-
-## Active Caching and Invalidation
-
-The cache we have created above is a "passive" cache; it only pulls data from the data source as needed, and has no knowledge of if and when data from the data source has actually changed, so it must rely on timer-based expiration to periodically retrieve possibly updated data. This means that it is possible that the cache may have stale data for a while (if the underlying data has changed, but the cached data hasn't expired), and the cache may have to refresh more than necessary if the data source data hasn't changed. Consequently it can be significantly more effective to implement an "active" cache, in which the data source is monitored and notifies the cache when any data changes. This ensures that when data changes, the cache can immediately load the updated data, and unchanged data can remain cached much longer (or indefinitely).
-
-### Invalidate
-
-One way to provide more active caching is to specifically invalidate individual records. Invalidation is useful when you know the source data has changed, and the cache needs to re-retrieve data from the source the next time that record is accessed. This can be done by executing the `invalidate()` method on a resource. For example, you could extend a table (in your resources.js) and provide a custom POST handler that does invalidation:
-
-```javascript
-const { MyTable } = tables;
-export class MyTableEndpoint extends MyTable {
-	async post(data) {
-		if (data.invalidate)
-			// use this flag as a marker
-			this.invalidate();
-	}
+type BreedCache @table(expiration: 3600) @export {
+  id: ID @primaryKey
 }
 ```
+Here, `expiration: 3600` means cached entries expire after an hour. Harper will automatically evict old data and refresh it when needed. By exporting this table, you instantly get a REST API for it at `http://localhost:9926/BreedCache/`.
```

Comment:
Harper just marks data as expired. It will not optimistically refresh data. The next request from a user to an expired record will result in the refetch and cache being restored.